SWIG/options.i

```diff
@@ -2289,4 +2289,102 @@ class BinomialDoubleBarrierEnginePtr : public boost::shared_ptr<PricingEngine> {
     }
 };
 
+
+// Swing option
+
+%{
+using QuantLib::VanillaSwingOption;
+typedef boost::shared_ptr<Instrument> VanillaSwingOptionPtr;
+%}
+
+%rename(VanillaSwingOption) VanillaSwingOptionPtr;
+class VanillaSwingOptionPtr : public boost::shared_ptr<Instrument> {
+  public:
+    %extend {
+        VanillaSwingOptionPtr(
+            const boost::shared_ptr<Payoff>& payoff,
+            const boost::shared_ptr<Exercise>& ex,
+            Size minExerciseRights, Size maxExerciseRights) {
+            
+            const boost::shared_ptr<SwingExercise> swingExercise =
+                 boost::dynamic_pointer_cast<SwingExercise>(ex);
+            QL_REQUIRE(swingExercise, "Swing exercise required");            
+            
+            return new VanillaSwingOptionPtr(
+                new VanillaSwingOption(payoff, swingExercise, 
+                    minExerciseRights, maxExerciseRights));
+        }
+    }
+};
+
+// Swing engines
+
+%{
+using QuantLib::FdSimpleBSSwingEngine;
+using QuantLib::FdSimpleExtOUJumpSwingEngine;
+typedef boost::shared_ptr<PricingEngine> FdSimpleExtOUJumpSwingEnginePtr;
+typedef boost::shared_ptr<PricingEngine> FdSimpleBSSwingEnginePtr;
+typedef boost::shared_ptr<FdSimpleExtOUJumpSwingEngine::Shape> CurveShapePtr;
+%}
+
+%rename(FdSimpleBSSwingEngine) FdSimpleBSSwingEnginePtr;
+class FdSimpleBSSwingEnginePtr : public boost::shared_ptr<PricingEngine> {
+  public:
+    %extend {
+        FdSimpleBSSwingEnginePtr(
+            const GeneralizedBlackScholesProcessPtr& process,
+            Size tGrid = 50, Size xGrid = 100,
+            const FdmSchemeDesc& schemeDesc = FdmSchemeDesc::Douglas()) {
+            boost::shared_ptr<GeneralizedBlackScholesProcess> bsProcess =
+                 boost::dynamic_pointer_cast<GeneralizedBlackScholesProcess>(
+                                                                     process);
+            QL_REQUIRE(bsProcess, "Black-Scholes process required");
+            
+            return new FdSimpleBSSwingEnginePtr(
+                new FdSimpleBSSwingEngine(bsProcess,tGrid, xGrid, schemeDesc));
+        }
+    }
+};
+
+%rename(CurveShape) CurveShapePtr;
```

Comment:
@klausspanderen : since this is a `vector<pair<Time, Real> >` I'd export it as such so that, e.g., from Python one can pass directly a list of pairs.  It probably won't make a lot of difference for Java or Scala: I think replacing `new CurveShape()` with `new DoublePairVector()` and `curveShape.add(t, gs)` with `curveShape.add(DoublePair(t, gs))` should work.

Do you mind if I do the change?

Reply:
Cool, please go ahead. I wasn't sure how to do it.

Reply:
it should be `curveShape.add(`**`new`**` DoublePair(t, gs))`